sdcm/utils/common.py

```diff
@@ -1796,14 +1796,60 @@ def get_ami_images(branch: str, region: str, arch: AwsArchType) -> list:
 
     for ami in amis:
         tags = {i['Key']: i['Value'] for i in ami.tags}
-        rows.append(["AWS", ami.name, ami.image_id, ami.creation_date, tags.get(
+        rows.append(["AWS", ami.name, ami.image_id, tags.get("Name"), ami.creation_date, tags.get(
             'build-id', tags.get("build_id"))[:6], tags.get('arch'), tags.get('ScyllaVersion')])
 
     return rows
 
 
+def get_ec2_image_name_tag(ami: EC2Image) -> str:
+    if ami.tags:
+        for tag in ami.tags:
+            if tag["Key"] == "Name":
+                return tag["Value"]
+    return ""
+
+
+def convert_name_to_ami_if_needed(ami_id_param: str, region_names: list[str],) -> str:
+    """
+    convert image name in ami_id param to ami_ids
+    Firstly trying to find name in 'tag:Name' - for ScyllaDB images case
+    then in 'name' - for ScyllaDB images case for public images like Ubuntu
+
+    usage example:
+    convert_name_to_ami_if_needed('scylla-5.1.4-x86_64-2023-01-22T17-58-39Z' ,
+                                  ['eu-west-1', 'eu-west-2', 'us-west-2', 'us-east-1', 'eu-north-1', 'eu-central-1'])
+    returns: 'ami-042cf1bc21e30ce60 ami-0370fbb289d8310d2 ami-09808043b7fcdb244
+              ami-00c1a16b5136fbb7c ami-0226185e7d7951b6a ami-0bc7811c417b0eb09'
+
+     convert_name_to_ami_if_needed('ubuntu/images/hvm-ssd/ubuntu-jammy-22.04-amd64-server-20221206' ,
+                                   'eu-west-1', 'eu-west-2', 'us-west-2', 'us-east-1', 'eu-north-1', 'eu-central-1'])
+    returns: 'ami-026e72e4e468afa7b ami-01b8d743224353ffe ami-03f8756d29f0b5f21
+              ami-06878d265978313ca ami-03df6dea56f8aa618 ami-0039da1f3917fa8e3'
+    """
+    param_values = ami_id_param.split()
+    if len(param_values) == 1 and not param_values[0].startswith("ami-"):
```

Comment:
If we use a name like `ami-for-loader` then we will get incorrect behavior here.
Better to use regex here to detect the ami ID.

Then, it is not safe to assume that multiple values are only ami IDs. Need to verify it and look for images if we detect that `name` is provided instead of `ID`.

Reply:
I'm fine with not safe guarding from name prefix with `ami-`, we didn't checking guarding for that before either, it would fail when trying to create an instance with that, and that's fine.

as for supporting mix of names and id, I find it less relevant to the use cases where we need the names, i.e. loader/monitor images, the whole point is that we want to stop specifying AMI ids per region, and again there was no validation before, and it would fail on creation, and that's fine.

Reply:
@fruch 
It is not only about failures, it is also about flexibility. If we want to use not latest ID then we specify it explicitly.
For the moment it is not supported.

Reply:
> @fruch It is not only about failures, it is also about flexibility. If we want to use not latest ID then we specify it explicitly. For the moment it is not supported.

can you elaborate on which case isn't supported now ? specifying ami id would still work, specifying it by name would now work, what's not going to work ?

Reply:
`'some-name-first-element ami-fooid` and other `multiple` combinations

Reply:
I'm o.k. with define that anything that start with `ami-` we treat as ami id. 
complicating this for this case isn't a waste of time from my POV.

I think a note about it in the configuration option such suffice, if one would run into this he shoule be able to change the name quite easily. and I don't think any formal AMI of anything would ever start like that...

Reply:
Ok, if it is intentional, let it be so.